src/iter.rs

```diff
@@ -88,7 +89,7 @@ impl<'a, S> Iterator for IterPinRef<'a, S> {
             let next = (*self.task).spin_next_all(self.pending_next_all, Relaxed);
             self.task = next;
             self.len -= 1;
-            Some(Pin::new_unchecked(stream))
+            Some((task.id, Pin::new_unchecked(stream)))
```

Comment:
I'm not 100% sure that it's safe for us to still access `task` here, so let's instead have:
```rust
let id = (*self.task).id;
```
directly next to `let stream` further up, and then just use `id` directly here.